From 308691e03cc6312bde3409b346df3156d34db0fe Mon Sep 17 00:00:00 2001
From: Mohammad Nweider <nweiderm@amazon.com>
Date: Wed, 25 Oct 2017 16:26:54 +0000
Subject: [PATCH] Ticket 49401 - Fix compiler incompatible-pointer-types
 warnings

Bug Description: vs->sorted was integer pointer in older versions,
    but now it's size_t pointer, this is causing compiler warnings
    during the build

Fix Description: use size_t pointers instead of integer pointers for vs->sorted and sorted2

Review By: mreynolds

Signed-off-by: Mark Reynolds <mreynolds@redhat.com>
(cherry picked from commit 52ba2aba49935989152010aee0d40b01d7a78432)
---
 ldap/servers/slapd/valueset.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/ldap/servers/slapd/valueset.c b/ldap/servers/slapd/valueset.c
index 1c1bc150a..dc0360738 100644
--- a/ldap/servers/slapd/valueset.c
+++ b/ldap/servers/slapd/valueset.c
@@ -680,7 +680,7 @@ valueset_array_purge(const Slapi_Attr *a, Slapi_ValueSet *vs, const CSN *csn)
     int nv = 0;
     int numValues = 0;
     Slapi_Value **va2 = NULL;
-    int *sorted2 = NULL;
+    size_t *sorted2 = NULL;
 
     /* Loop over all the values freeing the old ones. */
     for(i = 0; i < vs->num; i++)
@@ -750,7 +750,7 @@ valueset_array_purge(const Slapi_Attr *a, Slapi_ValueSet *vs, const CSN *csn)
         if(vs->sorted) {
             /* Let's allocate va2 and sorted2 */
             va2 = (Slapi_Value **) slapi_ch_malloc( (numValues + 1) * sizeof(Slapi_Value *));
-            sorted2 = (int *) slapi_ch_malloc( (numValues + 1)* sizeof(int));
+            sorted2 = (size_t *) slapi_ch_malloc( (numValues + 1)* sizeof(size_t));
         }
 
         /* I is the index for the *new* va2 array */
@@ -804,7 +804,7 @@ valueset_array_purge(const Slapi_Attr *a, Slapi_ValueSet *vs, const CSN *csn)
 
     /* We still have values but not sorted array! rebuild it */
     if(vs->num > VALUESET_ARRAY_SORT_THRESHOLD && vs->sorted == NULL) {
-        vs->sorted = (int *) slapi_ch_malloc( vs->max* sizeof(int));
+        vs->sorted = (size_t *) slapi_ch_malloc( vs->max* sizeof(size_t));
         valueset_array_to_sorted(a, vs);
     }
 #ifdef DEBUG
-- 
2.13.6

